exporter/humioexporter/config.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2021, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package humioexporter
+
+import (
+	"errors"
+	"net/url"
+	"path"
+
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+)
+
+const (
+	basePath         = "api/v1/ingest/"
+	unstructuredPath = basePath + "humio-unstructured"
+	structuredPath   = basePath + "humio-structured"
+)
+
+// Humio configuration settings specific to logs
+type LogsConfig struct {
+	// The name of a custom log parser to use, if no parser is associated with the ingest token
+	LogParser string `mapstructure:"log_parser"`
+}
+
+// Humio configuration settings specific to traces
+type TracesConfig struct {
+	// Whether to use Unix timestamps, or to fall back to ISO 8601 formatted strings
+	UnixTimestamps bool `mapstructure:"unix_timestamps"`
+
+	// The time zone to use when representing timestamps in Unix time
+	TimeZone string `mapstructure:"timezone"`
+}
+
+// Humio configuration settings
+type Config struct {
+	// Inherited settings
+	configmodels.ExporterSettings `mapstructure:",squash"`
+	confighttp.HTTPClientSettings `mapstructure:",squash"`
+	exporterhelper.QueueSettings  `mapstructure:"sending_queue"`
+	exporterhelper.RetrySettings  `mapstructure:"retry_on_failure"`
+
+	//Ingest token for identifying and authorizing with a Humio repository
+	IngestToken string `mapstructure:"ingest_token"`
+
+	// Endpoint for the unstructured ingest API, created internally
+	unstructuredEndpoint *url.URL
+
+	// Endpoint for the structured ingest API, created internally
+	structuredEndpoint *url.URL
+
+	// Key-value pairs used to target specific data sources for storage inside Humio
+	Tags map[string]string `mapstructure:"tags,omitempty"`
+
+	// Whether this exporter should automatically add the service name as a tag
+	DisableServiceTag bool `mapstructure:"disable_service_tag"`
+
+	// Configuration options specific to logs
+	Logs LogsConfig `mapstructure:"logs"`
+
+	// Configuration options specific to traces
+	Traces TracesConfig `mapstructure:"traces"`
+}
+
+// Ensure that a valid configuration has been provided, such that we can fail early
+func (c *Config) sanitize() error {
```

Comment:
If you name it like `Validate()`, you might take advantage of the collector validating this automatically for you in the near future.

See https://github.com/open-telemetry/opentelemetry-collector/issues/2541

Reply:
Please address this.